reference/normalize.go

```diff
@@ -56,6 +56,35 @@ func ParseNormalizedNamed(s string) (Named, error) {
 	return named, nil
 }
 
+// NormalizeImageRef normalizes the image reference following the docker convention. This is added
+// mainly for backward compatibility.
+// The reference returned can only be either tagged or digested. For reference contains both tag
+// and digest, the function returns digested reference, e.g. docker.io/library/busybox:latest@
+// sha256:7cc4b5aefd1d0cadf8d97d4350462ba51c694ebca145b08d7d41b41acc8db5aa will be returned as
+// docker.io/library/busybox@sha256:7cc4b5aefd1d0cadf8d97d4350462ba51c694ebca145b08d7d41b41acc8db5aa.
+func NormalizeImageRef(ref string) (Named, error) {
```

Comment:
Maybe we should just call this `ParseDockerRef` since `Normalize` means something different here. 

Just as a note, I want to split the reference package out of this repository and explicitly have a `Docker` referencing package for all projects to use which make use of this referencing logic. Ironically this reference package is least useful to this repository since the reference format used by the registry can be vastly simplified as a superset of what Docker uses. I don't have time to refactor this, but it is not harmful to have this functionality along side all the other Docker-specific normalization functions.

Reply:
I like that name better 👍